instrumentation/kafka-clients/src/test/java/brave/kafka/clients/ITKafkaTracing.java

```diff
@@ -45,97 +40,57 @@
 import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.TestName;
-import org.junit.rules.TestRule;
-import org.junit.rules.TestWatcher;
-import org.junit.runner.Description;
 import zipkin2.DependencyLink;
 import zipkin2.Span;
+import zipkin2.Span.Kind;
 import zipkin2.internal.DependencyLinker;
 
-import static brave.kafka.clients.BaseTracingTest.takeSpan;
 import static brave.kafka.clients.KafkaTags.KAFKA_TOPIC_TAG;
 import static brave.messaging.MessagingRequestMatchers.channelNameEquals;
 import static brave.messaging.MessagingRequestMatchers.operationEquals;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.groups.Tuple.tuple;
 
-public class ITKafkaTracing {
-
-  String TEST_KEY = "foo";
-  String TEST_VALUE = "bar";
-
-  /**
-   * See brave.http.ITHttp for rationale on using a concurrent blocking queue eventhough some calls,
-   * like consumer operations, happen on the main thread.
-   */
-  BlockingQueue<Span> consumerSpans = new LinkedBlockingQueue<>();
-  BlockingQueue<Span> producerSpans = new LinkedBlockingQueue<>();
-
-  KafkaTracing consumerTracing = KafkaTracing.create(Tracing.newBuilder()
-    .localServiceName("consumer")
-    .currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
-      .addScopeDecorator(StrictScopeDecorator.create())
-      .build())
-    .spanReporter(consumerSpans::add)
-    .build());
-  KafkaTracing producerTracing = KafkaTracing.create(Tracing.newBuilder()
-    .localServiceName("producer")
-    .currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
-      .addScopeDecorator(StrictScopeDecorator.create())
-      .build())
-    .spanReporter(producerSpans::add)
-    .build());
-
+public class ITKafkaTracing extends ITKafka {
   @ClassRule
   public static KafkaJunitRule kafkaRule = new KafkaJunitRule(EphemeralKafkaBroker.create());
-  @Rule
-  public TestName testName = new TestName();
+
+  @Rule public TestSpanReporter producerReporter = new TestSpanReporter();
```

Comment:
@anuraaga this is using multiple reporters.. indeed efforts like this are piecemeal as things come up. Mainly because it is so much work to refactor all the tests across all modules. Once things settle, and things stop changing, we can think hard about cleaner design with the benefit of seeing warts bit by bit.